app/Http/Controllers/AdminController.php

```diff
@@ -7,29 +7,108 @@
 use App\Models\User;
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Session;
 
 class AdminController extends Controller
 {
     /**
      * Display a listing of the resource.
      */
-    public function index()
-    {
-       return view('Admin.user.admin.allAdmin');
+    // public function index(Request $request)
+    // {
+
+   
+    // $slug = $request->get('slug');
+    // $email = $request->get('email');
+    // $active = $request->has('status') && $request->get('status') === 'active';
+    // $blocked = $request->has('status') && $request->get('status') === 'blocked';
+
+    // $query = Admin::query()
+    //     ->with('userAdmin')
+    //     ->leftJoin('users', 'users.id', '=', 'admins.user_id');
+
+    // if ($slug) {
+    //     $query->where('users.slug', $slug);
+    // }
+
+    // if ($email) {
+    //     $query->where('users.email', $email);
+    // }
+
+    // if ($active) {
+    //     $query->where('users.active', 1);
+    // }
+
+    // if ($blocked) {
+    //     $query->where('users.active', 0);
+    // }
+
+    // // $query->where('users.position', 'admin'); // Filter for admin position
+    // $admins = $query->select('admins.*')->paginate(25)->appends(['slug' => $slug, 'email' => $email]);
+
+        
+    //     return view('Admin.user.admin.allAdmin', compact('admins'));
+    // }
+
+    public function index(Request $request)
+{
+  $admins = $this->searchWith($request);
+
+  return view('Admin.user.admin.allAdmin', compact('admins'));
+}
+
+
+private function searchWith(Request $request)
+{
+    $slug = $request->get('slug');
+    $email = $request->get('email');
+    $active = $request->has('status') && $request->get('status') === 'active';
+    $blocked = $request->has('status') && $request->get('status') === 'blocked';
+
+        $query = Admin::query()
+            ->with('userAdmin')
+            ->leftJoin('users', 'users.id', '=', 'admins.user_id');
+
+    if ($slug) {
+        $query->where('users.slug', $slug);
     }
 
+        if ($email) {
+            $query->where('users.email', $email);
+        }
+
+        if ($active) {
+            $query->where('users.active', 1);
+        }
+
+        if ($blocked) {
+            $query->where('users.active', 0);
+        }
+
+    return $query->select('admins.*')->paginate(25)->appends(['slug' => $slug, 'email' => $email]);
+}
+        
+
+        
+        
+        
+    
     /**
      * Show the form for creating a new resource.
      */
     public function create()
     {
-        return view('Admin.user.admin.addAdmin');
+        $admins = Admin::all();
+  
+        return view('Admin.user.admin.addAdmin',compact('admins'));
     }
 
     /**
      * Store a newly created resource in storage.
      */
     public function store(UserRequest $request)
+    
```

Comment:
your store function is a little bit missy

When you create a new admin, you also create a new user, but your function is not creating a new user.
Also why you don't use messages `error messages` and `success messages` look at the article for example to see how to do so.